pages/restake/lrt_developers/lrt-vault.mdx

```diff
@@ -0,0 +1,300 @@
+- MetaMask wallet connected to Tangle Network
+- Some test tokens for deployment (on testnet)
+
+Install Foundry:
+
+```bash
+curl -L https://foundry.paradigm.xyz | bash
+foundryup
+```
+
+## Understanding the Components
+
+The Tangle Liquid Restaking implementation consists of the following key components:
+
+1. **Vault Contract (TangleLiquidRestakingVault)**: An ERC4626-compliant vault that:
+
+   - Manages deposits and withdrawals
+   - Implements reward distribution with index-based accounting
+   - Handles delegation through the MultiAssetDelegation precompile
+   - Provides liquid token representation of staked assets
+
+2. **MultiAssetDelegation Wrapper**: Interfaces with Tangle's MultiAssetDelegation precompile at `0x0000000000000000000000000000000000000822`
+
+Core features include:
+
+```solidity
+contract TangleLiquidRestakingVault is ERC4626, Owned, TangleMultiAssetDelegationWrapper {
+    /// @notice Scale factor for reward index calculations
+    uint256 private constant REWARD_FACTOR = 1e18;
+
+    /// @notice Tracks user's reward snapshot for a token
+    struct RewardSnapshot {
+        uint256 rewardIndex;    // Global index at snapshot creation
+        uint256 timestamp;      // When snapshot was created
+        uint256 shareBalance;   // User's share balance at snapshot
+        uint256 lastRewardIndex; // Index in rewardTimestamps array
+        uint256 pendingRewards; // Unclaimed rewards at snapshot
+    }
+
+    /// @notice Operator address for delegation
+    bytes32 public operator;
+
+    /// @notice Blueprint selection for delegation and exposure
+    uint64[] public blueprintSelection;
+}
+```
+
+## Step 1: Setting Up the Project
+
+First, clone the reference implementation:
+
+```bash
+git clone https://github.com/tangle-network/lrt
+cd lrt
+forge soldeer update -d
+```
+
+## Step 2: Core Implementation Details
+
+### Reward Distribution System
+
+The vault implements a sophisticated reward distribution system using index-based accounting:
+
+```solidity
+/// @notice Calculate pending rewards for a user
+function _calculatePendingRewards(address user, address token) internal view returns (uint256) {
+    RewardSnapshot memory snapshot = userSnapshots[user][token];
+    RewardData storage rData = rewardData[token];
+
+    // Calculate rewards since last snapshot
+    uint256 indexDelta = rData.index - snapshot.rewardIndex;
+    // Use mulDivUp for final reward calculation to ensure no dust is left behind
+    uint256 newRewards = snapshot.shareBalance.mulDivUp(indexDelta, REWARD_FACTOR);
+    uint256 totalRewards = newRewards + snapshot.pendingRewards;
+
+    return totalRewards;
+}
+```
+
+### Delegation Management
+
+The vault handles delegation through the MultiAssetDelegation precompile, managing deposits and withdrawals:
+
+```solidity
+/// @notice Deposit assets into vault and delegate
+function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
```

Comment:
If these functions are actually implemented in the repo then why not use the GitHub file reader component and view them through that?